packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1251,18 +1321,19 @@ export class SnapController extends BaseController<
     version?: string,
   ): Promise<FetchSnapResult> {
     try {
-      if (snapId.startsWith(SnapIdPrefixes.local)) {
-        return this._fetchLocalSnap(snapId.replace(SnapIdPrefixes.local, ''));
-      } else if (snapId.startsWith(SnapIdPrefixes.npm)) {
-        return this._fetchNpmSnap(
-          snapId.replace(SnapIdPrefixes.npm, ''),
-          version,
-        );
+      const snapPrefix = snapIdToSnapPrefix(snapId);
+      switch (snapPrefix) {
+        case SnapIdPrefixes.local:
+          return this._fetchLocalSnap(snapId.replace(SnapIdPrefixes.local, ''));
+        case SnapIdPrefixes.npm:
+          return this._fetchNpmSnap(
+            snapId.replace(SnapIdPrefixes.npm, ''),
+            version,
+          );
+        default:
+          // This whill fail to compile if the above switch is not fully exhaustive
+          return assertExhaustive(snapPrefix);
```

Comment:
Should we just throw `Invalid snap id` or something similar here instead?

Reply:
(as we did before below)

Reply:
Doesn't matter, won't reach it anyways

Reply:
Yeah, TypeScript should guarantee that we'll fail at compile-time before that ever happens.